hang_man.py

```diff
@@ -0,0 +1,148 @@
+import os
+import sys
+import copy
+import string
+import random
+import argparse
+import urllib.request
+
+
+class Hangman:
+
+    def __init__(self, min_limit):  # Creating init method that takes in max word limit as perimeter
```

Comment:
dont get me wrong, your commitment to the commenting is impressive, but overkill. In the future you dont need to comment every line. You should comment functions on what they do and a few explanations throughout the code to maybe explain an complex code block.